treeherder/model/models.py

```diff
@@ -1450,15 +1497,120 @@ class TextLogError(models.Model):
     step = models.ForeignKey(TextLogStep, related_name='errors')
     line = models.TextField()
     line_number = models.PositiveIntegerField()
+    failure_line = models.OneToOneField(FailureLine,
+                                        related_name="text_log_error",
+                                        null=True)
+    # Note that the case of best_classification = None and best_is_verified = True
+    # has the special semantic that the line is ignored and should not be considered
+    # for future autoclassifications.
+    best_classification = models.ForeignKey("ClassifiedFailure",
+                                            related_name="best_for_errors",
+                                            null=True,
+                                            db_index=True,
```

Comment:
db_index is not necessary here, as django automatically indexes by column where a foreign key is specified.